Create sub packages for each optimiser

Vaibhavdixit02 opened the pull request:
This PR changes the structure of the repo and stops use of Requires based loading, the supported optimiser backends will each now be a sub package.

ChrisRackauckas commented:
And we should make sure to add the tests for them too

Vaibhavdixit02 commented:
Yes I am doing that as well

ChrisRackauckas commented:
Can you get to a place where tests pass and we can merge? Please don't do this in a way where every package has to be moved before the first PR can merge.

Vaibhavdixit02 commented:
>Can you get to a place where tests pass and we can merge? Please don't do this in a way where every package has to be moved before the first PR can merge.

I am not sure I understood? So right now I have moved Flux and BBO and their tests will pass (I haven't added it to CI and adjusted existing test files) so are you suggesting to do this incrementally in subsequent PRs?

ChrisRackauckas commented on src/GalacticOptim.jl:
```diff
@@ -21,11 +21,6 @@ include("function/function.jl")
-    @require Evolutionary="86b6b26d-c046-49b6-aa0b-5f0f74682bd6" include("solve/evolutionary.jl")
-    @require Flux="587475ba-b771-5e3f-ad9e-33799f191a9c" include("solve/flux.jl")
-    @require GCMAES="4aa9d100-eb0f-11e8-15f1-25748831eb3b" include("solve/gcmaes.jl")
     @require MathOptInterface="b8f27783-ece8-5eb3-8dc8-9495eed66fee" include("solve/moi.jl")
```
forgot to remove

ChrisRackauckas commented on src/GalacticOptim.jl:
```diff
@@ -21,11 +21,6 @@ include("function/function.jl")
-    @require GCMAES="4aa9d100-eb0f-11e8-15f1-25748831eb3b" include("solve/gcmaes.jl")
     @require MathOptInterface="b8f27783-ece8-5eb3-8dc8-9495eed66fee" include("solve/moi.jl")
     @require MultistartOptimization="3933049c-43be-478e-a8bb-6e0f7fd53575" include("solve/multistartoptimization.jl")
     @require NLopt="76087f3c-5699-56af-9a33-bf431cd00edd" include("solve/nlopt.jl")
```
forgot to remove

Vaibhavdixit02 replied:
Yup, will be done

ChrisRackauckas commented:
All of the system-wide issues seem to be fixed, so I'm merging so that each of the per-package issues that are remaining can be handled separately.
